### DISCUSSION TITLE
Implementation of request Search Merge in 0.11.1

### DISCUSSION THREAD

**Discussion by wineternity** [milvus-io/milvus#6303] [Q&A and General discussion]
Hello, just a question of the request merge in request scheduler.
In milvus 1.x version, the SearchReq will be merged to process to enhance the performence. May I ask is it possible to port it into the 0.11.x version? And why 0.11.x not implement this?

Thanks for answer.

**Comment by xiaofan-luan** [comment]
> Hello, just a question of the request merge in request scheduler.
> In milvus 1.x version, the SearchReq will be merged to process to enhance the performence. May I ask is it possible to port it into the 0.11.x version? And why 0.11.x not implement this?
> 
> Thanks for answer.

Hi wineternity, 1.x is the continuity of 0.1x, it is called 1.x simply because it is much stable and we believe it's ready to be a formal release.  you can easily upgrade from 0.1x to 1.x, so please upgrade if you are looking for a stable version.

2.x is totally a different story.We change pretty much everything in the system. If you already put Milvus into production then I guess it's your call to upgrade to 2.x or stay in 1.x for a while.

**Reply by wineternity** [reply]
I know 1.x is from 0.10.x , 0.11.x is totally a different version from 0.10.x(or namely 1.x now). 0.11.x is the experimental version of Milvus which support scalar filtering.
I am talking about is there any problem to add the request merging functionality to 0.11.x.  Is there exist any  skill problem or bugs?
https://zhuanlan.zhihu.com/p/148274838
I just want to test whether it could add this functionality to 0.11.x

**Reply by xiaofan-luan** [reply]
yep, i don't think there are any issue to backport this feature.  Currently  2.0  is our priority and we should put the request merging into 2.0. Feel free to open an issue and done the backport if you are interested? we may have one release of 1.x in next month

**Reply by wineternity** [reply]
OK , thanks for reply.

**Comment by yhmo** [comment]
The 0.11.0 was abandoned before we migrate the search-combine enhancement from 0.10.x.
Since 0.11.0 is multi-fields and DSL search, the parameters of SearchRequest is totally different to 0.10.x. To implement it, you need to rewrite the combine logic.
This is SearchRequest construction of 0.10.x & 1.x:
```
static BaseRequestPtr
Create(const std::shared_ptr<milvus::server::Context>& context, const std::string& collection_name,
           const engine::VectorsData& vectors, int64_t topk, const milvus::json& extra_params,
           const std::vector<std::string>& partition_list, const std::vector<std::string>& file_id_list,
           TopKQueryResult& result);
```
And this is 0.11.0:
```
static BaseReqPtr
Create(const ContextPtr& context, const query::QueryPtr& query_ptr, const milvus::json& json_params,
           engine::snapshot::FieldElementMappings& collection_mappings, engine::QueryResultPtr& result);
```

So If you want to implement the search combination in 0.11.0, two steps you need to do:
1. Create a new class SearchCombineRequest under the /core/src/server/delivery/request, implement the logic like the 1.x
2. Implement the SearchReqStrategy::ReScheduleQueue in the /core/src/server/delivery/strategy/SearchReqStrategy.cpp
